src/IntegralArrays.jl

```diff
@@ -1,5 +1,78 @@
 module IntegralArrays
 
 # package code goes here
+using Images, ClosedIntervals
+
+import Base: size, getindex, linearindexing
+
+export IntegralArray, ±, ..
+
+"""
+```
+integral_array = IntegralArray(A)
+```
+
+Returns the integral array of an array. The integral array is calculated by assigning
+to each cell the sum of all cells above it and to its left, i.e. the rectangle from
+(1, 1) to the pixel. An integral array is a data structure which helps in efficient 
+calculation of sum of pixels in a rectangular subset of an array. 
+
+```
+sum = integral_array[ytop..ybot, xtop..xbot]
+sum = integral_array[y ± σy, x ± σx]
+sum = integral_array[CartesianIndex(y, x) ± σ]
+```
+
+The sum of a window in an array can be directly calculated using four array references of 
+the integral array, irrespective of the size of the window, given the `yrange` and `xrange` 
+of the window. Given an integral array - 
+        
+        A - - - - - - B -
+        - * * * * * * * -
+        - * * * * * * * -
+        - * * * * * * * -
+        - * * * * * * * -
+        - * * * * * * * -
+        C * * * * * * D - 
+        - - - - - - - - -
+
+The sum of pixels in the area denoted by * is given by S = D + A - B - C.
+"""
+immutable IntegralArray{T, N, A} <: AbstractArray{T, N}
+	data::A
+end
+
+function IntegralArray(array::AbstractArray)
+    integral_array = Array{Images.accum(eltype(array))}(size(array))
+    sd = coords_spatial(array)
+    cumsum!(integral_array, array, sd[1])
+    for i = 2:length(sd)
+        cumsum!(integral_array, integral_array, sd[i])
+    end
+    IntegralArray{eltype(array), ndims(array), typeof(array)}(integral_array)
+end
+
+..(x, y) = ClosedInterval(x, y)
+±(x, y) = ClosedInterval(x + y, x - y)
+
+function ±(x::CartesianIndex, y)
+	ClosedInterval(x + y, x - y)
```

Comment:
maybe `map(ClosedInterval, (x-y).I, (x+y).I)` or something? (I don't have that package installed.)